src/action.ts

```diff
@@ -134,7 +145,12 @@ async function linkAllFilesInMetafile(
 				e === "NOT_FOUND" ? "TORRENT_NOT_FOUND" : e,
 			);
 		}
-		sourceRoot = join(downloadDirResult.unwrapOrThrow(), searchee.name);
+		sourceRoot = join(
+			downloadDirResult.unwrapOrThrow(),
+			searchee.files.length === 1
+				? searchee.files[0].path
+				: searchee.name,
+		);
```

Comment:
We initially did have this as isSingleFileTorrent, I believe we changed it to accommodate the counter-example you listed if memory serves.

@ShanaryS will need to confirm that my memory is not failing me. There were dozens of iterations of some things here :P

Reply:
This check was added because of your counterexample. There would still only be a single file in that just with a path of `Movie/Movie.mkv`. We could add it on searchee but it's a very simple check.

Reply:
But a multi file torrent with one file wouldn't be changed by subfolder layout since it's already in a subfolder, right? I'm not sure I understand.